cornflow-server/cornflow/commands/views.py

```diff
@@ -68,3 +41,108 @@ def register_views_command(external_app: str = None, verbose: bool = False):
             current_app.logger.info("No new endpoints to be registered")
 
     return True
+
+
+def load_changes_to_db(views_to_delete, views_to_register, views_to_update):
+    if len(views_to_register) > 0:
+        db.session.bulk_save_objects(views_to_register)
+    if len(views_to_update) > 0:
+        db.session.bulk_update_mappings(ViewModel, views_to_update)
+    if len(views_to_delete) > 0:
+        for view_id in views_to_delete:
+            view_to_delete = ViewModel.get_one_object(idx=view_id)
+            if view_to_delete:
+                view_to_delete.delete()
+    try:
+        db.session.commit()
+    except IntegrityError as e:
+        db.session.rollback()
+        current_app.logger.error(f"Integrity error on views register: {e}")
+    except DBAPIError as e:
+        db.session.rollback()
+        current_app.logger.error(f"Unknow error on views register: {e}")
+
+
+def get_views_to_update_and_delete(
+    all_resources_to_register_views_endpoints, views_registered_urls_all_attributes
+):
+    views_to_delete = []
+    views_to_update = []
+    # Check if views have the same name but different url_rule or description
+    for view_name, view_attrs in views_registered_urls_all_attributes.items():
+        if view_name in all_resources_to_register_views_endpoints.keys():
+            new_endpoint = all_resources_to_register_views_endpoints[view_name]
+            if (
+                view_attrs["url_rule"] != new_endpoint["url_rule"]
+                or view_attrs["description"] != new_endpoint["description"]
+            ):
+                views_to_update.append(
+                    {
+                        "id": view_attrs["id"],
+                        "name": view_name,
+                        "url_rule": new_endpoint["url_rule"],
+                        "description": new_endpoint["description"],
+                    }
+                )
+        else:
+            views_to_delete.append(view_attrs["id"])
+    return views_to_delete, views_to_update
+
+
+def get_views_to_register(resources_to_register, views_registered):
+    views_to_register = [
+        ViewModel(
+            {
```

Comment:
I think that you should just create the dict here and not apply view_model.

And just return views_to_register in the function